components/eam/src/physics/cam/gw_common.F90

```diff
@@ -742,4 +746,1472 @@ subroutine gw_drag_prof(ncol, ngwv, src_level, tend_level, do_taper, dt, &
+  dtaux3d_ls=dtaux3_ls(:ncol,pver:1:-1),dtauy3d_ls=dtauy3_ls(:ncol,pver:1:-1),&
+  dtaux3d_bl=dtaux3_bl(:ncol,pver:1:-1),dtauy3d_bl=dtauy3_bl(:ncol,pver:1:-1),&
+  dtaux3d_ss=dtaux3_ss(:ncol,pver:1:-1),dtauy3d_ss=dtauy3_ss(:ncol,pver:1:-1),&
+  dtaux3d_fd=dtaux3_fd(:ncol,pver:1:-1),dtauy3d_fd=dtauy3_fd(:ncol,pver:1:-1),&
+  dusfcg_ls=dusfc_ls(:ncol),dvsfcg_ls=dvsfc_ls(:ncol),&
+  dusfcg_bl=dusfc_bl(:ncol),dvsfcg_bl=dvsfc_bl(:ncol),&
+  dusfcg_ss=dusfc_ss(:ncol),dvsfcg_ss=dvsfc_ss(:ncol),&
+  dusfcg_fd=dusfc_fd(:ncol),dvsfcg_fd=dvsfc_fd(:ncol),&
+  xland=cam_in%landfrac,br=state%ribulk(:ncol),&
+  var2d=sgh(:ncol),oc12d=state%oc(:ncol),&
+  oa2d=state%oadir(:ncol,:),ol2d=state%ol(:ncol,:),&
+  znu=etamid(pver:1:-1),dz=dz(:ncol,pver:1:-1),pblh=pblh(:ncol),&
+  cp=cpair,g=gravit,rd=rair,rv=rh2o,ep1=zvir,pi=pi,bnvbg=nm(:ncol,pver:1:-1),&
+  dt=dtime,dx=dx,dy=dy,&
+  kpbl2d=kpbl2d_reverse_in,gwd_opt=0,&
+  ids=1,ide=ncol,jds=0,jde=0,kds=1,kde=pver, &
+  ims=1,ime=ncol,jms=0,jme=0,kms=1,kme=pver, &
+  its=1,ite=ncol,jts=0,jte=0,kts=1,kte=pver, &
+  gwd_ls=gwd_ls,gwd_bl=gwd_bl,gwd_ss=gwd_ss,gwd_fd=gwd_fd )
+
+end subroutine gw_oro_interface
+
+!==========================================================================
+
+function pblh_get_level_idx(height_array,pblheight)
+  implicit none
+  real(r8),intent(in),dimension(pver) :: height_array
+  real(r8),intent(in) :: pblheight
+  integer :: pblh_get_level_idx
+  !local 
+  integer :: k
+  logical :: found
+
+  pblh_get_level_idx = -1
+  found=.false.
+  !get the pblh level index and return
+  do k = 1, pver
+    if((pblheight >= height_array(k+1).and.pblheight <height_array(k)))then
+      pblh_get_level_idx =  k+1
+      found=.true.
+      return
+    endif
+  enddo
+
+end function
+
+!==========================================================================
+
+subroutine dxygrid(dx,dy,theta_in,dxy)
+
+  IMPLICIT NONE
+  real(r8),intent(in) :: dx,dy,theta_in
+  real(r8),intent(out):: dxy
+  !local variables
+  real(r8) :: rad,theta,theta1
+
+  rad=4.0_r8*atan(1.0_r8)/180.0_r8
+  theta1=MOD(theta_in,360._r8)
+  !set negative axis into 0~360
+  if (theta1.ge.-360._r8.and.theta1.lt.0._r8) then
+  theta1=theta1+360._r8
+  endif
+  !in case the angle is not into the judgement
+  theta=theta1
+  !transform of angle into first quadrant
+  if      (theta1.ge.  0._r8.and.theta1.lt. 90._r8) then
+  theta=theta1
+  else if (theta1.gt. 90._r8.and.theta1.lt.180._r8) then
+  theta=(180._r8-theta1)
+  else if (theta1.gt.180._r8.and.theta1.lt.270._r8) then
+  theta=(theta1-180._r8)
+  else if (theta1.gt.270._r8.and.theta1.lt.360._r8) then
+  theta=(360._r8-theta1)
+  else if (theta1.eq.90._r8.or.theta1.eq.270._r8) then
+  theta=90._r8
+  else if (theta1.eq.0._r8.or.theta1.eq.180._r8) then
+  theta=0._r8
+  endif
+
+  !get dxy
+  if   (theta.ge. 0._r8.and.theta.lt.atan2(dy,dx)/rad) then
+  dxy=dx/cos(theta*rad)
+  else if (theta.ge.atan2(dy,dx)/rad.and.theta.le.90._r8)then
+  dxy=dy/sin(theta*rad)
+  endif
```

Comment:
another indentation issue, but also this routine could benefit from a description of what it's calculating. It's not immediately clear to me what "dxy" is. I'm guessing it's like a solid angle sort of thing with units of radians squared?

Reply:
dxy is the model grid length (in meter) in the wind direction. It is added in description.